pandas/core/ops/common.py

```diff
@@ -93,11 +93,7 @@ def get_op_result_name(left, right):
     name : object
         Usually a string
     """
-    if isinstance(right, (ABCSeries, ABCIndex)):
-        name = _maybe_match_name(left, right)
-    else:
-        name = left.name
-    return name
+    return left.name
```

Comment:
for devs: alternatively could modify `_maybe_match_name()` to return a.left if it can't find matching names or is_matching_na  == 1.